### PR TITLE
CATL-1248: Fix draft PDF activity form

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview

This PR fixes the following issues:

* We were displaying the email form for PDF draft activities.
* The email and PDF drafts should be displayed in view mode from the activity panel and in edit mode when resuming their draft.

## Before & After

<table>
  <tr>
    <th></th>
    <th>Before</th>
    <th>After</th>
  </tr>
  <tr>
    <th>Contact Activity Tab</th>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <th>Case Activity Tab</th>
    <td></td>
    <td></td>
  </tr>
</table>

## Technical Details

The issue has always existed so it's not a regression. These are the list of changes to fix the issue:

* We needed to check if a given activity was a draft, but also belonged to a particular activity type (not specific) so we refactored  `check-if-draft-activity.factory.js` for this purpose.
* Since we need to display different forms for the email and for the PDF letters we split the previous `DraftEmailOrPdfActivityForm ` into `DraftEmailActivityForm` and `DraftPdfActivityForm`.
* We have added a new `canChangeStatus` property to each one of the activity forms. The email and PDF drafts have this value set to false. This property is used in the activity panel to determine if the dropdown to change the status should be displayed or not.
* The activity panel was also using `checkIfDraftEmailOrPDFActivity` service, so we updated it so it implements the new `checkIfDraftActivity` service instead.
* For the activity panel we also pass the `{ action: 'view' }` option to the activity form. This ensures that the form is displayed in view mode when within the panel, but opens in edit/create mode when resuming the activity draft flow.
* The `CRM.url` spy function was updated so it does not break tests where URL parameters are not defined in a particular order. Note: We use the result of `CRM.url` instead of spying on it because we make sure that whoever is implementing this function is calling it + returning the right value.

### CODE REVIEW

**Comment by jamienovick** [comment]
@deb1990  I don't think is correct. 

We shouldn't be showing the editor in the pane as it wont fit as you can see. It should be in the popup modal form only.

i.e. when I create a new PDF it open the modal. I then click save draft it creates the draft activity. I think click the item in the list it should show the normal view activity page with the resume draft option. I then click resume draft and the form should open on modal.

Does that make sense?

Thanks

J

**Comment by deb1990** [comment]
Thanks @jamienovick . It seemed like the screenshots were taken in edit mode, but I just verified that by default Draft PDF is opening in Edit mode. So yes we do need to fix it.

**Comment by reneolivo** [comment]
thanks a lot @jamienovick !

**Comment by jamienovick** [comment]
@reneolivo @deb1990 @pubudup sorry another thing here. 

I think the activity status should be hidden when you view the draft emails. We shouldn't allow users to change the status from draft to another status. Their options are to a) Create the letters (which can be done from the modal window) or b) delete the draft. They shouldn't be able to change the status from draft.

The same should apply to draft emails as well please.

Thanks

J

**Comment by deb1990** [comment]
@reneolivo Found a bug, when you Resume and change something for a PDF document, and resave as Draft. We get duplicate Activities.

File: ang/civicase/activity/activity-forms/services/draft-email-activity-form.service.js [hunk 1]
```diff
@@ -29,11 +29,12 @@
      */
     function getActivityFormUrl (activity, optionsWithoutDefaults) {
       var options = _.defaults({}, optionsWithoutDefaults, { action: 'add' });
+      var assigneeContactId = _.first(activity.assignee_contact_id);
```
[the remark this answers is not in the document]
Comment (reneolivo) [reply ang/civicase/activity/activity-forms/services/draft-email-activity-form.service.js]: @tunbola when we pass multiple contacts both forms break in different ways:

Email:
Screen Shot 2020-03-20 at 12 01 49 PM
(the activity is not shown)

PDF:
Screen Shot 2020-03-20 at 12 02 24 PM

It seems for the PDF the redirection is not done right.

[the remark this answers is not in the document]
Comment (tunbola) [reply ang/civicase/activity/activity-forms/services/draft-email-activity-form.service.js]: I guess we'll need to investigate this then in another ticket.

**Pull request merged**